File: app/src/main/java/org/tvheadend/tvhclient/data/entity/ServerStatus.java
package org.tvheadend.tvhclient.data.entity;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;

@Entity(tableName = "server_status", indices = {@Index(value = {"id", "connection_id"}, unique = true)})
public class ServerStatus {

    @PrimaryKey(autoGenerate = true)
    private int id;
    @ColumnInfo(name = "connection_id")
    public int connectionId;
    @ColumnInfo(name = "connection_name")
    private String connectionName;
    @ColumnInfo(name = "server_name")
    private String serverName;
    @ColumnInfo(name = "server_version")
    private String serverVersion;
    private String webroot;
    private long time = 0;
    private String timezone;
    @ColumnInfo(name = "gmt_offset")
    private int gmtoffset = 0;
    @ColumnInfo(name = "htsp_version")
    private int htspVersion = 13;
    @ColumnInfo(name = "free_disc_space")
    private long freeDiskSpace = 0;
    @ColumnInfo(name = "total_disc_space")
    private long totalDiskSpace = 0;
    @ColumnInfo(name = "channel_tag_id")
    private int channelTagId = 0;
    @ColumnInfo(name = "playback_server_profile_id")
    private int playbackServerProfileId = 0;
    @ColumnInfo(name = "recording_server_profile_id")
    private int recordingServerProfileId = 0;
    @ColumnInfo(name = "casting_server_profile_id")
    private int castingServerProfileId = 0;
    @ColumnInfo(name = "playback_transcoding_profile_id")
    private int playbackTranscodingProfileId = 0;
    @ColumnInfo(name = "recording_transcoding_profile_id")
    private int recordingTranscodingProfileId = 0;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getConnectionId() {
        return connectionId;
    }

    public void setConnectionId(int connectionId) {
        this.connectionId = connectionId;
    }

    public String getConnectionName() {
        return connectionName;
    }

    public void setConnectionName(String connectionName) {
        this.connectionName = connectionName;
    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
    }

    public String getTimezone() {
        return timezone;
    }

    public void setTimezone(String timezone) {
        this.timezone = timezone;
    }

    public int getGmtoffset() {
        return gmtoffset;
    }

    public void setGmtoffset(int gmtoffset) {
        this.gmtoffset = gmtoffset;
    }

    public int getHtspVersion() {
        return htspVersion;
    }

    public void setHtspVersion(int htspVersion) {
        this.htspVersion = htspVersion;
    }

    public long getFreeDiskSpace() {
        return freeDiskSpace;
    }

    public void setFreeDiskSpace(long freeDiskSpace) {
        this.freeDiskSpace = freeDiskSpace;
    }

    public long getTotalDiskSpace() {
        return totalDiskSpace;
    }

    public void setTotalDiskSpace(long totalDiskSpace) {
        this.totalDiskSpace = totalDiskSpace;
    }

    public int getChannelTagId() {
        return channelTagId;
    }

    public void setChannelTagId(int channelTagId) {
        this.channelTagId = channelTagId;
    }

    public int getPlaybackServerProfileId() {
        return playbackServerProfileId;
    }

    public void setPlaybackServerProfileId(int playbackServerProfileId) {
        this.playbackServerProfileId = playbackServerProfileId;
    }

    public int getRecordingServerProfileId() {
        return recordingServerProfileId;
    }

    public void setRecordingServerProfileId(int recordingServerProfileId) {
        this.recordingServerProfileId = recordingServerProfileId;
    }

    public int getCastingServerProfileId() {
        return castingServerProfileId;
    }

    public void setCastingServerProfileId(int castingServerProfileId) {
        this.castingServerProfileId = castingServerProfileId;
    }

    public int getPlaybackTranscodingProfileId() {
        return playbackTranscodingProfileId;
    }

    public void setPlaybackTranscodingProfileId(int playbackTranscodingProfileId) {
        this.playbackTranscodingProfileId = playbackTranscodingProfileId;
    }

    public int getRecordingTranscodingProfileId() {
        return recordingTranscodingProfileId;
    }

    public void setRecordingTranscodingProfileId(int recordingTranscodingProfileId) {
        this.recordingTranscodingProfileId = recordingTranscodingProfileId;
    }

    public String getServerName() {
        return serverName;
    }

    public void setServerName(String serverName) {
        this.serverName = serverName;
    }

    public String getServerVersion() {
        return serverVersion;
    }

    public void setServerVersion(String serverVersion) {
        this.serverVersion = serverVersion;
    }

    public String getWebroot() {
        return webroot;
    }

    public void setWebroot(String webroot) {
        this.webroot = webroot;
    }
}

File: data/src/main/java/org/tvheadend/data/entity/ServerStatus.kt
package org.tvheadend.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

data class ServerStatus(

        var id: Int = 0,
        var connectionId: Int = 0,
        var connectionName: String? = null,
        var serverName: String? = null,
        var serverVersion: String? = null,
        var webroot: String? = null,
        var time: Long = 0,
        var timezone: String? = null,
        var gmtoffset: Int = 0,
        var htspVersion: Int = 13,
        var freeDiskSpace: Long = 0,
        var totalDiskSpace: Long = 0,
        var channelTagId: Int = 0,
        var htspPlaybackServerProfileId: Int = 0,
        var httpPlaybackServerProfileId: Int = 0,
        var recordingServerProfileId: Int = 0,
        var seriesRecordingServerProfileId: Int = 0,
        var timerRecordingServerProfileId: Int = 0,
        var castingServerProfileId: Int = 0,
        var playbackTranscodingProfileId: Int = 0,
        var recordingTranscodingProfileId: Int = 0
)

@Entity(tableName = "server_status", indices = [Index(value = ["id", "connection_id"], unique = true)])
internal data class ServerStatusEntity(

        @PrimaryKey(autoGenerate = true)
        var id: Int = 0,
        @ColumnInfo(name = "connection_id")
        var connectionId: Int = 0,
        @ColumnInfo(name = "connection_name")
        var connectionName: String? = null,
        @ColumnInfo(name = "server_name")
        var serverName: String? = null,
        @ColumnInfo(name = "server_version")
        var serverVersion: String? = null,
        var webroot: String? = null,
        var time: Long = 0,
        var timezone: String? = null,
        @ColumnInfo(name = "gmt_offset")
        var gmtoffset: Int = 0,
        @ColumnInfo(name = "htsp_version")
        var htspVersion: Int = 13,
        @ColumnInfo(name = "free_disc_space")
        var freeDiskSpace: Long = 0,
        @ColumnInfo(name = "total_disc_space")
        var totalDiskSpace: Long = 0,
        @ColumnInfo(name = "channel_tag_id")
        var channelTagId: Int = 0,
        @ColumnInfo(name = "playback_server_profile_id")
        var htspPlaybackServerProfileId: Int = 0,
        @ColumnInfo(name = "http_playback_server_profile_id")
        var httpPlaybackServerProfileId: Int = 0,
        @ColumnInfo(name = "recording_server_profile_id")
        var recordingServerProfileId: Int = 0,
        @ColumnInfo(name = "series_recording_server_profile_id")
        var seriesRecordingServerProfileId: Int = 0,
        @ColumnInfo(name = "timer_recording_server_profile_id")
        var timerRecordingServerProfileId: Int = 0,
        @ColumnInfo(name = "casting_server_profile_id")
        var castingServerProfileId: Int = 0,
        @ColumnInfo(name = "playback_transcoding_profile_id")
        var playbackTranscodingProfileId: Int = 0,
        @ColumnInfo(name = "recording_transcoding_profile_id")
        var recordingTranscodingProfileId: Int = 0
) {
    companion object {
        fun from(serverStatus: ServerStatus): ServerStatusEntity {
            return ServerStatusEntity(
                    serverStatus.id,
                    serverStatus.connectionId,
                    serverStatus.connectionName,
                    serverStatus.serverName,
                    serverStatus.serverVersion,
                    serverStatus.webroot,
                    serverStatus.time,
                    serverStatus.timezone,
                    serverStatus.gmtoffset,
                    serverStatus.htspVersion,
                    serverStatus.freeDiskSpace,
                    serverStatus.totalDiskSpace,
                    serverStatus.channelTagId,
                    serverStatus.htspPlaybackServerProfileId,
                    serverStatus.httpPlaybackServerProfileId,
                    serverStatus.recordingServerProfileId,
                    serverStatus.seriesRecordingServerProfileId,
                    serverStatus.timerRecordingServerProfileId,
                    serverStatus.castingServerProfileId,
                    serverStatus.playbackTranscodingProfileId,
                    serverStatus.recordingTranscodingProfileId)
        }
    }

    fun toServerStatus(): ServerStatus {
        return ServerStatus(id, connectionId, connectionName, serverName, serverVersion, webroot, time, timezone, gmtoffset, htspVersion, freeDiskSpace, totalDiskSpace, channelTagId, htspPlaybackServerProfileId, httpPlaybackServerProfileId, recordingServerProfileId, seriesRecordingServerProfileId, timerRecordingServerProfileId, castingServerProfileId, playbackTranscodingProfileId, recordingTranscodingProfileId)
    }
}
